merkle-tree/indexed/tests/tests.rs

```diff
@@ -536,3 +542,251 @@ pub fn functional_non_inclusion_test() {
         .verify_non_inclusion_proof(&non_inclusion_proof)
         .unwrap();
 }
+
+#[derive(Debug, Default, Clone, Copy, BorshSerialize, BorshDeserialize)]
+pub struct RawIndexedElement {
+    pub value: [u8; 32],
+    pub next_index: usize,
+    pub next_value: [u8; 32],
+    pub index: usize,
+    pub next_value_index: usize,
+    pub next_value_next_value: [u8; 32],
+    pub next_value_path: [[u8; 32]; 10],
+}
+
+/// Check whether low element has changed
+/// - possibilities:
+///   1. check Merkle proof of -> too expensive costs like 20k CU
+///   2. check whether the array contains the index of the low element if yes then check the next value
+///   3. insert the index in to hash set that stores key: index value: next_value, can be overwritten if capacity requires,
+///         the hashset needs to be able to hold the same element multiple times though, does it?
+///         maybe not, maybe we can just say that we allow every element once and when we use it then we overwrite it immediately instead of overwriting an expired element
```

Comment:
Can we trim the comments length to ~100 characters per line? Otherwise it's hard to read (I often have multiple vertical panes of code open).

Reply:
this one is rambling anyway will remove or at least rewrite